beacon-chain/rpc/validator_server_test.go

```diff
@@ -238,3 +173,105 @@ func TestValidatorCommitteeAtSlot_Ok(t *testing.T) {
 		t.Errorf("Shard for validator at index 31 should be 2, received %d", res.Shard)
 	}
 }
+
+func TestNextEpochCommitteeAssignment_CantFindValidatorIdx(t *testing.T) {
+	db := internal.SetupDB(t)
+	defer internal.TeardownDB(t, db)
+
+	if err := db.SaveState(&pbp2p.BeaconState{ValidatorRegistry: []*pbp2p.Validator{}}); err != nil {
+		t.Fatalf("could not save state: %v", err)
+	}
+	vs := &ValidatorServer{
+		beaconDB: db,
+	}
+	req := &pb.ValidatorIndexRequest{
+		PublicKey: []byte{'A'},
+	}
+	want := fmt.Sprintf("can't find validator index for public key %#x", req.PublicKey)
+	if _, err := vs.NextEpochCommitteeAssignment(context.Background(), req); !strings.Contains(err.Error(), want) {
+		t.Errorf("Expected %v, received %v", want, err)
+	}
+}
+
+func TestNextEpochCommitteeAssignment_OK(t *testing.T) {
+	db := internal.SetupDB(t)
+	defer internal.TeardownDB(t, db)
+	genesis := b.NewGenesisBlock([]byte{})
+	if err := db.SaveBlock(genesis); err != nil {
+		t.Fatalf("Could not save genesis block: %v", err)
+	}
+	state, err := genesisState(params.BeaconConfig().DepositsForChainStart)
+	if err != nil {
+		t.Fatalf("Could not setup genesis state: %v", err)
+	}
+	if err := db.UpdateChainHead(genesis, state); err != nil {
+		t.Fatalf("Could not save genesis state: %v", err)
+	}
+	vs := &ValidatorServer{
+		beaconDB: db,
+	}
+
+	// Test the first validator in registry.
+	var pubKey [96]byte
+	copy(pubKey[:], []byte(strconv.Itoa(0)))
+	req := &pb.ValidatorIndexRequest{
+		PublicKey: pubKey[:],
+	}
+	res, err := vs.NextEpochCommitteeAssignment(context.Background(), req)
+	if err != nil {
+		t.Errorf("Could not call next epoch committee assignment %v", err)
+	}
+	if res.Shard != 1 {
```

Comment:
I've been very hesitant to add conditions like these to our tests because a lot broke with randao and this assignment is random anyways. We shouldn't test behavior that is hard coded this way for random assignments

Reply:
Fair enough. I'll test for the acceptable range instead of an exact number